gulp-core-build-typescript/src/TypeScriptTask.ts

```diff
@@ -193,6 +174,47 @@ export class TypeScriptTask extends GulpTask<ITypeScriptTaskConfig> {
     throw 'Do not use mergeConfig with gulp-core-build-typescript';
   }
 
+  private _compileProject(gulp: gulpType.Gulp, tsCompilerOptions: ts.Settings, destDir: string,
+    allStreams: NodeJS.ReadWriteStream[], result: { errorCount: number }): void {
+    /* tslint:disable:typedef */
+    const plumber = require('gulp-plumber');
+    const sourcemaps = require('gulp-sourcemaps');
+    /* tslint:enable:typedef */
+
+    const tsProject: ts.Project = this._tsProject = this._tsProject || ts.createProject(tsCompilerOptions);
+
+    // tslint:disable-next-line:no-any
+    let tsResult: any = gulp.src(this.taskConfig.sourceMatch)
+      .pipe(plumber({
+        errorHandler: (): void => {
+          result.errorCount++;
+        }
+      }));
+
+    if (this.taskConfig.emitSourceMaps) {
+      tsResult = tsResult.pipe(sourcemaps.init());
+    }
+
+    tsResult = tsResult
+      .pipe(ts(tsProject, undefined, this.taskConfig.reporter));
+
+    // tslint:disable-next-line:typedef
+    let jsResult = (this.taskConfig.removeCommentsFromJavaScript
+      ? tsResult.js.pipe(require('gulp-decomment')({
+        space: !!this.taskConfig.emitSourceMaps
```

Comment:
A bit thrown off by the `space` option and what's happening here. Might be worth a comment.